code/modules/research/experimentor.dm

```diff
@@ -722,6 +728,165 @@
+		to_charge.charge = to_charge.maxcharge
+		// The device powered by the cell is assumed to be its location.
+		var/obj/device = to_charge.loc
+		// If it's not an object, or the loc's assigned power_store isn't the cell, undo.
+		if(!istype(device) || (device.get_cell() != to_charge))
+			device = to_charge
+		device.update_appearance(UPDATE_ICON|UPDATE_OVERLAYS)
+		to_chat(user, span_notice("[device] feels energized!"))
+		lightning_fx(device, 0.8 SECONDS)
+
+/obj/item/relic/proc/lightning_fx(atom/shocker, time)
+	var/lightning = mutable_appearance('icons/effects/effects.dmi', "electricity3", layer = ABOVE_MOB_LAYER)
+	shocker.add_overlay(lightning)
+	addtimer(CALLBACK(src, PROC_REF(cut_the_overlay), shocker, lightning), time)
+
+/obj/item/relic/proc/cut_the_overlay(atom/shocker, lightning)
+	shocker.cut_overlay(lightning)
+
+// Hugs/shakes everyone in range!
+/obj/item/relic/proc/hugger(mob/user)
+	var/list/mob/living/carbon/huggeds = oviewers(3, user)
+	for(var/mob/living/carbon/victim in huggeds)
+		victim.help_shake_act(user, force_friendly = TRUE)
+		new /obj/effect/temp_visual/heart(victim.loc)
+	if(length(huggeds))
+		to_chat(user, span_nicegreen("You feel friendly!"))
+	else
+		to_chat(user, pick(span_notice("You hug yourself, for some reason."), span_notice("You have a strange feeling for a moment, but then it passes.")))
+
+// Converts a 3x3 area into a random dimensional theme.
+/obj/item/relic/proc/dimensional_shift(mob/user)
+	var/datum/dimension_theme/shifter = pick(subtypesof(/datum/dimension_theme))
+	shifter = new shifter(src)
+	for(var/turf/shiftee in range(1, user))
+		shifter.apply_theme(shiftee, show_effect = TRUE)
+	qdel(shifter)
+	// prevent *total* spam conversion
+	min_cooldown += 2 SECONDS
+	max_cooldown += 2 SECONDS
+
+// Replaces your clothing with a random costume, and your ID with a cardboard one.
+// TODO: make them part of the same kit (lobster hat, lobster suit)
+/obj/item/relic/proc/disguiser(mob/user)
+	if(!iscarbon(user))
+		to_chat(user, span_notice("You have a strange feeling for a moment, but then it passes."))
+		return
+
+	if(prob(80)) // >:)
+		ADD_TRAIT(user, TRAIT_NO_JUMPSUIT, REF(src)) // prevent dropping pockets & belt
+
+	// magic trick!
+	playsound(get_turf(user), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	throw_smoke(get_turf(user))
+
+	// carbons always get a hat at least
+	var/mob/living/carbon/carbonius = user
+	//hat
+	var/obj/item/clothing/head/costume/disguise_hat = roll_costume(/obj/item/clothing/head/costume, HIDEMASK)
+	carbonius.dropItemToGround(carbonius.head)
+	carbonius.equip_to_slot_or_del(disguise_hat, ITEM_SLOT_HEAD)
+	if(!ishuman(carbonius))
+		to_chat(user, span_notice("You have a peculiar feeling for a moment, but then it passes."))
+		return
+
+	var/mob/living/carbon/human/humerus = carbonius
+	// uniform
+	var/obj/item/clothing/under/costume/disguise_uniform = roll_costume(/obj/item/clothing/under/costume)
+	humerus.dropItemToGround(humerus.w_uniform)
```

Comment:
Note `dropItemToGround` can fail, for nodrop items. Don't know if you care or not

Reply:
yeah i know